porcupine/plugins/highlight/tree_sitter_highlighter.py

```diff
@@ -150,6 +150,13 @@ def update_tags_of_visible_area_from_tree(self) -> None:
             #
             # There's a similar situation in rust: macro name is just an identifier in a macro_invocation
             if (
+                self._language_name == "python"
+                and node.type == "identifier"
+                and node.prev_sibling is not None
+                and node.prev_sibling.type in ("def", "class")
+            ):
+                type_name = f"{node.prev_sibling.type}_name"
```

Comment:
```suggestion
                # class Foo:  --> highlight Foo as a class_name
                type_name = f"{node.prev_sibling.type}_name"
```

I don't like this part of the highlighter in general. Apparently tree-sitter has some queries that seem to be like CSS selectors. I'll try them for this eventually.